src/pipe.js

```diff
@@ -30,5 +34,11 @@ var reverse = require('./reverse');
  *      squareThenDoubleThenTriple(5); //=> 150
  */
 module.exports = function pipe() {
-  return compose.apply(this, reverse(arguments));
+  return reduce(function(f, g) {
+    return curryN(f.length + g.length - 1, function() {
+      var fArgs = take(f.length, arguments);
+      var gArgs = drop(f.length, arguments);
+      return g.apply(this, prepend(f.apply(this, fArgs), gArgs));
+    });
+  }, identity, arguments);
```

Comment:
i don't object to the fold. i don't see why to curry inside every invocation of the accummulator. why not curry once after folding? that means you have to scan the list of functions to get the cumulative length, but that seems less onerous than re-currying every step. debugging through that stack trace will be insane